broadinstitute/leonardo
# GAWB-2665: Whitelisting Endpoints

Companion pull requests:
leonardo-configs: https://github.com/broadinstitute/firecloud-develop/pull/727
FiaB_lite: https://github.com/broadinstitute/firecloud-develop/pull/726
local: https://github.com/broadinstitute/firecloud-develop/pull/725

Non-cookie authed endpoints - get the email from the header and check if it's in the whitelist
Cookie authed enpoint - use the token in the browser to get user email from google. Cache the token/email pair.

To test:  
-  Build locally with the `gawb2665_local` branch of `firecloud-develop`
-  Auth as an email belonging to the whitelist, you should be able to access all the  non-cookie auth endpoints. For the cookie authed endpoint, try it out and then use the bearer token in the curl call and run the following in the browser console `javascript:document.cookie="FCToken=<TOKEN>"` Then, without refreshing the page, try the endpoint out again.
-  Auth as an email not belonging to the whitelist. You should receive a 401 error from both cookie and non-cookie endpoints (you'll have to re-add the token to the browser). 

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:


In all cases:

- [ ] Squash and merge; you can delete your branch after this
